src/main/java/org/valdroz/vscript/BaseNode.java

```diff
@@ -353,9 +353,40 @@ public Variant execute(VariantContainer variantContainer) {
                 if ((result == null || result.isNull()) && (valueSubstitution != null)) {
                     result = valueSubstitution.execute(variantContainer);
                 }
+
                 result = Variant.sanitize(result);
             }
             break;
+            case NT_MF_IF: {
+                if (parentRunBlock == null) {
+                    throw new UndefinedFunction(getName());
+                }
+                AbstractFunction function = parentRunBlock.resolveFunction(getName());
+                if (function == null) {
+                    throw new UndefinedFunction(getName());
+                }
+                LocalVariantContainer lvc = new LocalVariantContainer(variantContainer);
+                List<String> parameterNames = function.getParameterNames();
+                if (parameterNames.size() != 3) {
+                    throw new EvaluationException("Illegal Number of Paramters.");
```

Comment:
Error message does not provide sufficient detail to consumer to understand what is wrong`Illegal Number of Paramters.`

Reply:
updated Error message.